cardano-api/src/Cardano/Api/Typed.hs

```diff
@@ -1605,69 +1608,61 @@ parseMultiSigScript hms = case  all' hms <|> any' hms <|> atLeast hms of
-                                     error "atLeast multisig missing \"list\" key"
-                                   (Nothing, Just _) ->
-                                     error "atLeast multisig missing \"required\" key"
-                                   (Just (Number sci), Just (Array listVectorValue)) ->
-                                      case toBoundedInteger sci of
-                                        Just reqInt ->
-                                          Just $ fromJSONmOfnChecks
-                                                   (gatherMultiSigScripts listVectorValue)
-                                                   reqInt
-                                                   (Vector.length listVectorValue)
-                                        Nothing -> error $ "Error in multisig \"required\" key: "
-                                                         <> show sci <> " is not a valid Int"
-                                   (_, _) -> Nothing
-                     Just mss -> Just mss
-                 _ -> Nothing
-
-  -- To account for backwards compatibility
-  previousScriptSyntax :: HMS.HashMap Text Value -> Maybe MultiSigScript
-  previousScriptSyntax hm =
-        case HMS.lookup "paymentKeyHashes" hm of
-          Just (Array pkhListVectorValue) ->
-            case HMS.lookup "required" hm of
-              Nothing ->
-                error "atLeast multisig missing \"required\" key"
-              Just (Number sci) ->
-                case toBoundedInteger sci of
-                  Just reqInt ->
-                    Just $ fromJSONmOfnChecks
-                             (gatherMultiSigScripts pkhListVectorValue)
-                             reqInt
-                             (Vector.length pkhListVectorValue)
-                  Nothing -> error $ "Error in multisig \"required\" key: "
-                                   <> show sci <> " is not a valid Int"
-              _ -> Nothing
-          _ -> Nothing
+  atLeast hm =
+    case HMS.lookup "type" hm of
+      Just (String "atLeast") ->
+        case (HMS.lookup "required" hm, HMS.lookup "scripts" hm) of
+          (Nothing, Nothing) ->
+            error "atLeast multisig missing \"required\" and \"scripts\" keys"
+          (Just _, Nothing) ->
+            error "atLeast multisig missing \"required\" key"
+          (Nothing, Just _) ->
+            error "atLeast multisig missing \"scripts\" key"
+          (Just (Number sci), Just (Array listVectorValue)) ->
+             case toBoundedInteger sci of
+               Just reqInt ->
+                 Just $ fromJSONmOfnChecks
+                          (gatherMultiSigScripts listVectorValue)
+                          reqInt
+                          (Vector.length listVectorValue)
+               Nothing -> error $ "Error in multisig \"required\" key: "
+                                <> show sci <> " is not a valid Int"
+          (_, _) -> Nothing
+      _ -> Nothing
```

Comment:
Should probably also have test cases for the failure cases since there's a lot of custom logic in this area.